lib/grr.js

```diff
@@ -1399,13 +1403,244 @@ function stepListCRs(arg, cb) {
+                                    description: review_desc,
+                                    pattern: /[y|n]/,
+                                    message: 'y or n will do!',
+                                    required: true
+                                }
+                            }
+                        };
+                        prompt.colors = false;
+                        prompt.message = '';
+                        prompt.start();
+                        prompt.get(
+                            prompt_schema,
+                            function user_input(prompt_err, result) {
+                                if (prompt_err) {
+                                    next(new VError(
+                                        prompt_err,
+                                        'problem trying to prompt user'));
+                                    return;
+                                }
+                                if (result.answer == 'y') {
+                                    console.log(extsprintf(
+                                        'Granting a \'%s\' +1 to this review',
+                                        arg.reviewArg));
+                                    common.forkExecWaitAndLog({
+                                        argv: ['ssh', '-q',
+                                            '-o', 'StrictHostKeyChecking=no',
+                                            '-o', 'UserKnownHostsFile=/dev/null',
+                                            '-p', '29418',
+                                            arg.gerritUsername + '@cr.joyent.us',
+                                            'gerrit', 'review',
+                                            '--project', cr_res.project,
+                                            '--label', arg.reviewArg + '=+1',
+                                            cr_res.currentPatchSet.revision
+                                        ],
+                                        log: arg.log
+                                    // eslint-disable-next-line no-unused-vars
+                                    }, function (fork_err, review_res) {
+                                        if (fork_err) {
+                                            next(fork_err);
+                                            return;
+                                        }
+                                    });
+                                } else {
+                                    console.log('skipping this review.');
+                                }
+                                next();
+                        });
+                    } catch (parseErr) {
+                        next(parseErr);
+                        return;
+                    }
+                });
+            }
+        }
+    }, function (pipelineErr) {
+        if (pipelineErr) {
+            arg.log.trace({err: pipelineErr}, 'pipelineErr');
+            cb(pipelineErr);
```

Comment:
Does this work? It looks like `cb` is only called if there is a `pipelineErr`.

Reply:
Ah yep, you're right. As `stepReviewCRs` is the last function in the `grrReview()` vasync.pipeline, our lack of callback invocation didn't matter. As soon as I added another dummy step in that pipeline, we'd end up never calling it.

e.g.
```
function stepOodleBoodle(arg, cb) {
    console.log('hurrah, we made this function call.');
    cb();
    return;
}
.
. // then modify grrReview like this:
.
vasync.pipeline({arg: context, funcs: [
        stepGetGrrConfig,
        stepGetGrrCache,
        stepGetGerritUsername,
        stepQueryCRs,
        stepReviewCRs,
        stepOodleBoodle
    ]}, cb);
```

I've added the `cb(); return;` to `stepReviewCRS`, thanks for catching that (my async js still not up to scratch)